core/src/cluster_slots.rs

```diff
@@ -43,6 +47,12 @@ impl ClusterSlots {
         {
             let mut cluster_slots = self.cluster_slots.write().unwrap();
             *cluster_slots = cluster_slots.split_off(&(root + 1));
+            // Trimming is done at 2x size so that amortized it has a constant
+            // cost. The slots furthest away from the root are discarded.
+            if cluster_slots.len() > 2 * CLUSTER_SLOTS_TRIM_SIZE {
```

Comment:
This is a good start!

At 8 bytes a pubkey, 1000 validators, this can still get rather large though. Would be interesting to figure out how this cross cluster pollution is happening 🤔